community/cypher/frontend-3.0/src/main/scala/org/neo4j/cypher/internal/frontend/v3_0/ast/functions/Point.scala

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (c) 2002-2015 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.cypher.internal.frontend.v3_0.ast.functions
+
+import org.neo4j.cypher.internal.frontend.v3_0.ast.{Function, SimpleTypedFunction}
+import org.neo4j.cypher.internal.frontend.v3_0.symbols._
+
+case object Point extends Function with SimpleTypedFunction {
+  def name = "point"
+
+  val signatures = Vector(
+    Signature(argumentTypes = Vector(CTMap), outputType = CTGeometry)
```

Comment:
I only wanted to add to CTPoint type if I found it was actually necessary. Since we are going to have many geometry types, and even more DateTime types, I think keeping the range of types small (when it works) is perhaps a good idea. But we'll see how it goes. Perhaps specialized types are required at this level too.